modules/telco-ran-node-tuning-operator.adoc

```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+// * telco_ref_design_specs/ran/telco-ran-ref-du-components.adoc
+
+:_mod-docs-content-type: REFERENCE
+[id="telco-ran-node-tuning-operator_{context}"]
+= Node Tuning Operator
+
+New in this release::
+* No reference design updates in this release
+
+Description::
+You tune the cluster performance by link:https://docs.openshift.com/container-platform/latest/scalability_and_performance/cnf-create-performance-profiles.html[creating a performance profile].
+Settings that you configure with a performance profile include:
++
+* Selecting the realtime or non-realtime kernel.
+
+* Allocating cores to a reserved or isolated `cpuset`.
+{product-title} processes allocated to the management workload partition are pinned to reserved set.
+
+* Enabling kubelet features (CPU manager, topology manager, and memory manager).
+
+* Configuring huge pages.
+
+* Setting additional kernel arguments.
+
+* Setting per-core power tuning and max CPU frequency.
+
+Limits and requirements::
+
+The Node Tuning Operator uses the `PerformanceProfile` CR to configure the following for DU profile cluster nodes:
+
+* Sets the reserved `cpuset` to 4 or more 2-core hyperthreads on Intel 3rd Generation Xeon (Ice Lake) 2.20 GHz CPUs or better with firmware tuned for maximum performance.
+
+* Sets the reserved `cpuset` to include both hyperthread siblings for each included core. Unreserved cores are available as allocatable CPU for scheduling workloads.
+
+* Configures reserved and isolated CPUs to include all threads of all cores.
+
+* Sets Core 0 of each NUMA node to be included in the reserved CPU set.
+
+* Sets the huge page size to 1G.
+
+* Pins OpenShift pods that are configured by default as part of the management workload partition to the reserved cores.
+
+Engineering considerations::
+* Meeting the full performance metrics is based on use of the RT kernel.
++
+[NOTE]
+====
+You can use the non-RT kernel if required.
+====
+
+* The number of huge pages that you configure depends on the application workload requirements.
+Variation in this parameter is expected and allowed.
+
+* Variation is expected in the configuration of reserved and isolated CPU sets based on selected hardware and additional components in use on the system.
+Variation must still meet the specified limits.
+
+* Hardware without IRQ affinity support impacts isolated CPUs.
+To ensure that pods with guaranteed whole CPU QoS have full use of the allocated CPU, all hardware in the server must support IRQ affinity.
+For more information, see link:https://docs.openshift.com/container-platform/latest/scalability_and_performance/cnf-low-latency-tuning.html#about_irq_affinity_setting_cnf-master[About support of IRQ affinity setting].
```

Comment:
There is one additional consideration (late addition):
In 4.14 any PerformanceProfile CR configured on the cluster will cause NTO to automatically set all nodes of the cluster to use cgroups v1.

Reply:
Added this and added a link to https://docs.openshift.com/container-platform/4.14/nodes/clusters/nodes-cluster-cgroups-2.html#nodes-clusters-cgroups-2_nodes-cluster-cgroups-2. LMK if this works